### ISSUE TITLE
自行训练loss = nan

### ISSUE THREAD

**Issue by yangjinming1062** [libai3/masr#77]
使用的是网上能找到的aishell，prime，stcmd和thchs这四个数据集；ctcloss改成了torch.nn中的CTCLoss但是在传进去的out变量上套了一个log_softmax()；因为计算机显存的关系batch_size设置成了2。
现在的问题就是按这种配置第一个epoch都没跑完跑个一两万步就会出现nan值，不知何故。

**Comment by cloudthink** [comment]
我也换成了torch.nn.CTCLoss+log_softmax()，我把学习率调小（**0.1**），没遇到出现nan的情况。
但我现在遇到，用自己的数据（zh-CN）训练的时候，下面decode得到的out_strings都是空值。
eval()函数中，**out_strings**, out_offsets = decoder.decode(outs, out_lens)
![image](https://user-images.githubusercontent.com/15233548/62754362-16b5eb00-baa2-11e9-8f7a-8be3bb58f3cf.png)

**Comment by jet-yangqs** [comment]
请问你解决这个问题了吗？我训练的时候也遇到了相同的问题，能不能给点建议？谢谢

**Comment by yangjinming1062** [comment]
@jet-yangqs 有点记不清了，但是因为损失函数改了所以需要针对两个损失函数的差别进行一下修改
loss = ctcloss(nn.functional.log_softmax(out), y, out_lens, y_lens)这是第一处需要加个log_softmax其他还有哪些地方需要注意记不太清了；另外学习率也可以稍微调整一下

**Comment by feifaxiaoming** [comment]
@yangjinming1062  这块除了loss套上softmax之外，还需其他地方套入这个吗？

**Comment by feifaxiaoming** [comment]
你们都解决这个问题了吗，我的也出现nan了，第一个epoch走了1000来步，就有问题了。
